specification/dartLangSpec.tex

```diff
@@ -16276,38 +16279,29 @@ \subsection{Assignable Expressions}
 \end{grammar}
 
 \LMHash{}%
-An \Index{assignable expression} is either:
-\begin{itemize}
-\item An identifier.
-\item An invocation (possibly conditional) of a getter (\ref{getters})
-  or list access operator on an expression $e$.
-\item An invocation of a getter or list access operator on \SUPER.
-\end{itemize}
-
-\LMHash{}%
-An assignable expression of the form \id{} is evaluated as
-an identifier expression (\ref{identifierReference}).
+\BlindDefineSymbol{e, t}%
+Let $e$ be an \synt{assignableExpression}.
+Let $t$ be a term such that $e$ can be derived from
+\syntax{$t$ <assignableSelector>}.
+We say that $t$ is the
+\IndexCustom{receiver expression}{assignable expression!receiver expression}
+of the assignable expression $e$.
 
-%An assignable expression of the form \code{$e$.\id($a_1, \ldots,\ a_n$)}
-% is evaluated as a method invocation (\ref{methodInvocation}).
-
-\LMHash{}%
-An assignable expression of the form \code{$e$.\id} or \code{$e$?.\id}
-is evaluated as a property extraction (\ref{propertyExtraction}).
-
-\LMHash{}%
-An assignable expression of the form \code{$e_1$[$e_2$]}
-is evaluated as a method invocation of
-the operator method \code{[]} on $e_1$ with argument $e_2$.
-
-\LMHash{}%
-An assignable expression of the form \code{\SUPER.\id} is evaluated as
```

Comment:
We had special cases for `super` before, I think we still need them.
(Until we do something like https://github.com/dart-lang/language/blob/master/working/specification/compositional%20semantics.md).

Reply:
True! Fixed elsewhere.